### DISCUSSION TITLE
Torrent ratio calculation

### DISCUSSION THREAD

**Discussion by josephj11** [transmission/transmission#6441] [Q&A]
Empirically (I haven't read the code), it appears that ratios are calculated as uploaded/downloaded. I wonder why this was chosen. It would make more sense to me to use uploaded/torrent size.

Sometimes downloaded is significantly larger than torrent size even though there is no indication of bad blocks, etc., so I'm not clear on where this value comes from.

I don't frequently select only parts of a torrent to download (some files and not others), so I'm not sure how this affects these calculations.

**Comment by Pentaphon** [comment]
uploaded/downloaded is the total ratio. uploaded/torrent size is the per-torrent ratio.

**Reply by josephj11** [reply]
Yuh but, both seem to be calculated based on the total downloaded and not on the torrent size(s).

**Comment by GaryElshaw** [comment]
The through idea is determined from the ratio in preferences.

**Reply by josephj11** [reply]
Could you elaborate? I don't understand what you mean.

**Comment by tearfur** [comment]
I'd suggest discussing this in #6087.

**Reply by josephj11** [reply]
That request is a bit different than what I want, but I added a comment there anyway. Thanks.

**Reply by tearfur** [reply]
Yes, in fact, you and the issue owner want exact opposite behaviour. So it makes sense if we gather all discussions in one place.

**Reply by josephj11** [reply]
Yep. It figures. Everyone seems to look at things from their own unique perspective.

I have had quite a few "great" ideas over the years that no one else was interested in.

**Comment by YURIDUTY** [comment]
> Empirically (I haven't read the code), it appears that ratios are calculated as uploaded/downloaded. I wonder why this was chosen. It would make more sense to me to use uploaded/torrent size.
> 
> Sometimes downloaded is significantly larger than torrent size even though there is no indication of bad blocks, etc., so I'm not clear on where this value comes from.
> 
> I don't frequently select only parts of a torrent to download (some files and not others), so I'm not sure how this affects these calculations.



> Empirically (I haven't read the code), it appears that ratios are calculated as uploaded/downloaded. I wonder why this was chosen. It would make more sense to me to use uploaded/torrent size.
> 
> Sometimes downloaded is significantly larger than torrent size even though there is no indication of bad blocks, etc., so I'm not clear on where this value comes from.
> 
> I don't frequently select only parts of a torrent to download (some files and not others), so I'm not sure how this affects these calculations.

HI my downloading has taken a pause, but the torrent says it supposed to have more than 16 seeders.
it is like to most of the older torrents I try to download.
for new torrents uploaded in the website this week, I have no problems downloading them super fast.
why is my older torrents not downloading? does my ratio of .03 have anything to do with it? why they won't allow me to download?
please help

**Reply by josephj11** [reply]
If you set your upload ratio to zero or it is already met or if you set your upload speed limit very low, other peers will notice that and not give you data or restrict what speed they do offer. This is to discourage pure leechers who take, but do not give anything back to the swarm.

**Comment by YURIDUTY** [comment]
HI
Thank you for your response
how do i get my upload speed limit to be higher?
also, is it possible to buy credits to change my upload ratio?

any help is appreciated
thank you

On Fri, Jan 12, 2024 at 6:58 PM Joe ***@***.***> wrote:

> If you set your upload ratio to zero or it is already met or if you set
> your upload speed limit very low, other peers will notice that and not give
> you data or restrict what speed they do offer. This is to discourage pure
> leechers who take, but do not give anything back to the swarm.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/transmission/transmission/discussions/6441#discussioncomment-8115156>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFJGM5YEDMCZ5UEMA25NORLYOHEYRAVCNFSM6AAAAABBEFJMHWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DCMJVGE2TM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
YURI TIMG
Instructor/Coach
Applied Wing Chun Philippines

**Reply by josephj11** [reply]
TL;DR: Transmission can only slow things down. Going faster is usually between you and your ISP (internet service provider) and whomever provides the physical/radio connections between you and them.

Sometimes these services detect torrenting and throttle/reduce the bandwidth it can use. You can often avoid this by connecting to your ISP using a VPN (virtual private network) that encrypts all your traffic so your ISP has no idea what you're actually doing with it.

Transmission is a BitTorrent client. It allows you to leech and seed existing torrents and to create new torrent files. It does **_not_** control your access or bandwidth to any existing torrent or torrent aggregator, so there's nothing to pay or buy to use it or its full features.

You can (totally optionally) donate funds to the developers, but that will have no direct effect on how the program works or on how you can use it.

Transmission does have a couple of options to reduce your bandwidth globally or per torrent, but these are primarily useful if you have a low (or metered) bandwidth internet connection and need to reduce Transmission's bandwidth utilization so there's some left to do other things with other programs  like your browser.

Per torrent settings can be accessed by right clicking on a running/paused torrent in the Transmission main window, going to the Options tab. Deselect Honor global limits and select Limit download speed and/or Limit upload speed and set the desired associated speed in the fields to the right.

Global bandwidth limits can be set from the Transmission main window menu Edit->Preferences->Speed. These set the defaults for any torrent you add after changing these settings. They also configure the global slow down option which is activated when you select the Turtle icon in the Status bar of the main window. And there's a Speed settings icon to the left of that which is a shortcut for changing the global default settings.

You can also pause all or individual torrents if you want to. This can be handy if you want to test your internet connection speed with no existing load or if you want to stop everything and just start one torrent so it gets all the available bandwidth it can use and finishes uploading/downloading faster if the other peers it is connected to can provide more bandwidth that would have otherwise been shared with your other torrents.

**Comment by killemov** [comment]
The downlload/upload ratio calculation is weird because download maxes out at torrent size anyway so use the torrent size/upload calculation instead.

**Reply by josephj11** [reply]
Yeah. As of 4.0 that works. The downloaded figure often used to be significantly larger and was used in the ratios.
